staging/src/k8s.io/client-go/restmapper/discovery.go

```diff
@@ -152,6 +152,11 @@ func GetAPIGroupResources(cl discovery.DiscoveryInterface) ([]*APIGroupResources
 		}
 		// TODO track the errors and update callers to handle partial errors.
 	}
+
+	if apiGroups == nil {
```

Comment:
is it valid for the above to return nil and a non-error now? That doesn't seem right (and seems likely to break other callers)

Reply:
No, I didn't change the behavior of ServerGroups(). There was no clear contract stating `apiGroups==nil && err==nil` is impossible. I added this line to be more defensive.

Reply:
Ok, I thought this was necessary because of the changes in this PR